### PR TITLE
Add developer documentation for running a local CAS server

### PR DESCRIPTION
**Pull request by clokep**
This adds some documentation about running a local CAS server and hooking Synapse up to it. The docs are based on [our SAML docs](https://github.com/matrix-org/synapse/blob/229eb81498b0fe1da81e9b5b333a0285acde9446/docs/dev/saml.md). CAS is more annoying than SAML since Synapse needs to directly talk to the CAS server, so you can't do a lot of the stuff that samling does.

This uses a pretty simple package I found. I also pushed a git repo that has everything configured (which simplifies the configuration a bit): https://github.com/clokep/cas-test -- not sure if it is worth pointing to that or not though.

Here's the [rendered version](https://github.com/matrix-org/synapse/blob/clokep/cas-docs/docs/dev/cas.md).

This also makes a tweak to our SAML docs to note another parameter that must be set.

### CODE REVIEW

File: docs/dev/cas.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+2. Activate your virtualenv: `source /path/to/your/virtualenv/bin/activate`
+3. Install Django and django-mama-cas:
+   ```
+   python -m pip install "django<3" "django-mama-cas==2.4.0"
+   ```
+4. Create a Django project in the current directory:
+   ```
+   django-admin startproject cas_test .
+   ```
+5. Follow the [install directions](https://django-mama-cas.readthedocs.io/en/latest/installation.html#configuring) for django-mama-cas
```
Comment (anoadragon453) [inline docs/dev/cas.md]: I tried following these instructions and got stuck here. The linked instructions ask me to run `migrate`, but I'm not sure what this means. There certainly isn't a binary named `migrate` in my path.

Then I tried `python manage.py migrate`, thinking that might be it, but there's no file called `manage.py`.

What am I missing? :)

Comment (clokep) [reply docs/dev/cas.md]: There should be a file called `manage.py` after running the step above this (`django-admin startproject ...`)

Comment (anoadragon453) [reply docs/dev/cas.md]: Ah, that works! Would probably help if I didn't test this in such a cluttered folder.

Comment (clokep) [reply docs/dev/cas.md]: I would have liked the directions to be simpler, but...this seemed easier than a "full" CAS setup.

**Pull request merged**